### PR TITLE
MWPW-118345 - Tabs

### PR DESCRIPTION
**Pull request by ryanmparrish**
* Consonant Tabs

Resolves: [MWPW-118345](https://jira.corp.adobe.com/browse/MWPW-118345)

**Test URLs:**
- [Before](https://main--milo--adobecom.hlx.page/drafts/rparrish/tabs-content-model?martech=off)
- [Before](https://main--milo--adobecom.hlx.page/drafts/rparrish/tabs-content-model-dubs?martech=off)
- [After](https://tabbed-sections--milo--adobecom.hlx.page/drafts/rparrish/tabs-content-model?martech=off)
- [After](https://tabbed-sections--milo--adobecom.hlx.page/drafts/rparrish/tabs-content-model-dubs?martech=off)

### CODE REVIEW

File: libs/styles/styles.css [hunk 1]
```diff
@@ -211,6 +211,10 @@ picture.bg-img img {
     overflow: unset;
   }
 
+  header + main {
```
Comment (ryanmparrish) [inline libs/styles/styles.css:214]: Bonus Fix: site wide top alignment.

**Review by auniverseaway — APPROVED** [review]

**Comment by rgclayton** [comment]
@ryanmparrish I put together [a test page](https://tabbed-sections--milo--adobecom.hlx.page/drafts/rclayton/tabs-content-model-dubs) for me to tinker with. Check out tab 3. I'm noticing content is contained that would normally be fullwidth. What do you think?

**Comment by ryanmparrish** [comment]
@rgclayton - Good eye. I feel like we should remove that and the content its self should decide when it's contained and not the tabs styles. W/ that said do we then say any normal .content in the tab panel is contained?  Similar to how normal section text is handled.

**Comment by rgclayton** [comment]
> @rgclayton - Good eye. I feel like we should remove that and the content its self should decide when it's contained and not the tabs styles.

I feel the same. Let the authors decide if it should be contained or not. 

>  W/ that said do we then say any normal .content in the tab panel is contained? Similar to how normal section text is handled.

I would say yes if that section is authored like that. That way it operates the same as it would in a section outside of the tabs. Thoughts?

File: libs/blocks/tabs/tabs.js [hunk 2]
```diff
@@ -64,52 +53,106 @@ function changeTabs(e) {
...
+  const tabList = rows[0];
+  const tabId = `tabs-${initCount}`;
+  e.id = tabId;
+  tabList.classList.add('tabList');
+  tabList.setAttribute('role', 'tablist');
+  const tabListContainer = tabList.querySelector(':scope > div');
+  tabListContainer.classList.add('tabList-container');
+  const tabListItems = rows[0].querySelectorAll(':scope li');
+  if (tabListItems) {
+    let btnClass = [...e.classList].includes('quiet') ? 'heading-XS' : 'heading-XS'; // tabList size
```
Comment (meganthecoder) [inline libs/blocks/tabs/tabs.js:101]: nit: I'm not a huge fan of comments at the end of a line. I find them distracting and they get in the way but I'm curious what the Milo team thinks

File: libs/blocks/tabs/tabs.js [hunk 3]
```diff
@@ -64,52 +53,106 @@ function changeTabs(e) {
...
+      const tabName = getStringKeyName(item.textContent);
+      const tabBtnAttributes = {
+        role: 'tab',
+        class: btnClass,
+        id: `tab-${initCount}-${tabName}`,
+        tabindex: (i > 0) ? '0' : '-1',
+      }
+      const tabBtn = createTag('button', tabBtnAttributes);
+      tabBtn.setAttribute('aria-selected', (i === 0) ? 'true' : 'false');
+      tabBtn.setAttribute('aria-controls', `tab-panel-${initCount}-${tabName}`);
```
Comment (meganthecoder) [inline libs/blocks/tabs/tabs.js:112]: Doesn't `createTag` use `setAttribute` as well? Why not add them to the attributes object?

File: libs/blocks/tabs/tabs.js [hunk 4]
```diff
@@ -64,52 +53,106 @@ function changeTabs(e) {
...
-    const rowTitle = row.querySelector(':scope > div:nth-child(1)');
-    const tabBtnAttributes = {
-      role: 'tab',
-      class: btnClass,
-      id: `tab-${initCount}-${i}`,
-      tabindex: (i > 0) ? '0' : '-1',
+
+  // Tab Config
+  const config = { 'tab-id': initCount };
+  const configRows = e.querySelectorAll(':scope > div:not([class])');
```
Comment (meganthecoder) [inline libs/blocks/tabs/tabs.js:131]: Is there a more specific selector that's available?

File: libs/blocks/tabs/tabs.js (hunk 2 above)
Comment (ryanmparrish) [reply libs/blocks/tabs/tabs.js:101]: Agree. This comment is a bit redundant so i'll remove it.

File: libs/blocks/tabs/tabs.js (hunk 3 above)
Comment (ryanmparrish) [reply libs/blocks/tabs/tabs.js:112]: @meganthecoder - I tried that, but It doesn't seem to work w/ hyphenated attribute names. :( maybe theres a way?

File: libs/blocks/tabs/tabs.js (hunk 4 above)
Comment (ryanmparrish) [reply libs/blocks/tabs/tabs.js]: Actually, come to think of it, we don't even need another selector. We can just use the rows and splice the first one. This is updated.

File: libs/blocks/tabs/tabs.js (hunk 3 above)
Comment (meganthecoder) [reply libs/blocks/tabs/tabs.js:112]: I tried it and it seemed to work fine. Also you're missing a semi-colon. Do you have the ESLint extension set up?
```
     const tabBtnAttributes = {
        role: 'tab',
        class: btnClass,
        id: `tab-${initCount}-${tabName}`,
        tabindex: (i > 0) ? '0' : '-1',
        'aria-selected': (i === 0) ? 'true' : 'false',
        'aria-controls': `tab-panel-${initCount}-${tabName}`,
      };
```

**Comment by ryanmparrish** [comment]
Thanks for the feedback. 

@rgclayton  - I made the tabbed content not contained by default and build in a `contained` optional variant if the author did want to contain it. The tab content within a .section > .content does follow the same contained styles. This really should be set at the section level. Should we add a .contained style to section-metadata do you think?

@meganthecoder  - I tried it with a chart, it works as expected I think. Maybe take a peek and LMK if you could. I see that block does some parentElement section 2-up classifying. Im not sure how that works w/ it's respective container.

File: libs/blocks/tabs/tabs.js (hunk 3 above)
Comment (ryanmparrish) [reply libs/blocks/tabs/tabs.js]: Ok, thanks. I fixed that. I must have had tried incorrect syntax. Im lookin into my eslint IDE setting. Thanks

**Comment by meganthecoder** [comment]
@ryanmparrish Yeah the charts look great in there! As for the "up" situation, authors just need to utilize the multi-section per tab functionality. That's how it is for the charts anyway. LGTM https://tabbed-sections--milo--adobecom.hlx.page/drafts/methomas/tabs-content-model-dubs

**Comment by rgclayton** [comment]
> I made the tabbed content not contained by default and build in a `contained` optional variant if the author did want to contain it. The tab content within a .section > .content does follow the same contained styles. This really should be set at the section level. Should we add a .contained style to section-metadata do you think?

Ok cool. 
For the section container we have the no-block block (block with no class/name) to break out of the `contained` pattern for sections. Here's an [example in tab two ](https://tabbed-sections--milo--adobecom.hlx.page/drafts/rclayton/tabs-content-model-dubs)

**Review by rgclayton — APPROVED** [review]

**Pull request merged**